genai_factory/src/genai_factory/__main__.py

```diff
@@ -13,100 +13,93 @@
 # limitations under the License.
 #
 # main file with cli commands using python click library
-import importlib.util
-import sys
+
+import os
+import pathlib
 
 import click
-import uvicorn
+import dotenv
 
+from genai_factory import WorkflowServerConfig, workflow_server
 from genai_factory.api import router
-from genai_factory.chains.base import HistorySaver, SessionLoader
-from genai_factory.chains.refine import RefineQuery
-from genai_factory.chains.retrieval import MultiRetriever
-from genai_factory.config import config, username
-from genai_factory.workflows import AppServer
-
-default_graph = [
-    SessionLoader(),
-    RefineQuery(),
-    MultiRetriever(),
-    HistorySaver(),
+
+# Load the environment variables:
+_ROOT_PATH = pathlib.Path(__file__).parent.parent.parent
+dotenv.load_dotenv(os.environ.get("MLRUN_GENAI_ENV_PATH", str(_ROOT_PATH / ".env")))
+_USER_NAME = os.environ.get("MLRUN_GENAI_USER_NAME", "")
+_IS_LOCAL_CONFIG = os.environ.get(
+    "MLRUN_GENAI_IS_LOCAL_CONFIG", "0"
+).lower().strip() in [
+    "true",
+    "1",
 ]
+_CONFIG_PATH = os.environ.get("WORKFLOWS_CONFIG_PATH", None)
 
 
 @click.group()
 def cli():
     pass
 
 
-@click.command()
-@click.argument("workflow-name", type=str)
-@click.option("-p", "--path", type=str, default=None, help="Path to the workflow file")
-@click.option("-r", "--runner", type=str, default="fastapi", help="Runner to use")
+@click.command(help="Run a workflows server initialized in a given path.")
+@click.argument(
+    "workflows-path",
+    type=click.Path(exists=True, path_type=pathlib.Path),
+    # help="Path to the workflow file"
+)
 @click.option(
-    "-t",
-    "--workflow-type",
-    type=str,
-    default="application",
-    help="Type of the workflow",
+    "-c",
+    "--config-path",
+    type=click.Path(exists=True, path_type=pathlib.Path),
+    help="Path to a config file to use. Note: This option takes priority on both set configuration via the workflow "
+    "server instance in code, and via path from environment variable.",
+)
+@click.option(
+    "-r",
+    "--runner",
+    type=click.Choice(choices=["fastapi", "nuclio"]),
+    default="fastapi",
+    help="How to run the workflow/s.",
```

Comment:
well, the `runner` is actually a `deployer` if you think about it :D 
User can deploy locally using FastAPI or as a serverless function using Nuclio.

```suggestion
@click.option(
    "-d",
    "--deployer",
    type=click.Choice(choices=["fastapi", "nuclio"]),
    default="fastapi",
    help="How to deploy the workflow server with the added workflows in the given script.",
```